### PR TITLE
New CUDA kernels for binary broadcast, for: add, sub, mul, div, eq -operations

### PR DESCRIPTION
**Pull request by ap-hynninen**

### CODE REVIEW

**Comment by piiswrong** [comment]
Thanks!
You probably need to play with tol a little...

File: src/operator/tensor/elemwise_binary_broadcast_op.h [hunk 1]
```diff
@@ -57,6 +57,20 @@ inline bool BinaryBroadcastShape(const nnvm::NodeAttrs& attrs,
   return true;
 }
 
+#define NDIM_SWITCH(ndim, NDim, ...)  \
```
Comment (piiswrong) [inline src/operator/tensor/elemwise_binary_broadcast_op.h]: rename to BROADCAST_NDIM_SWITCH

File: tests/python/unittest/test_operator.py [hunk 2]
```diff
@@ -840,33 +840,74 @@ def test_convolution_grouping():
...
+        [[1, 1, 403, 17, 2], [1, 44, 403, 17, 2]],
+        [[1, 1, 65, 2, 22], [1, 1, 65, 1, 1]],
+        [[1, 24, 103, 17, 18], [1, 24, 1, 1, 1]],
+        [[1, 1, 1, 1, 2], [1, 24, 194, 50, 1]],
+        [[1, 1, 107, 84, 9], [1, 1, 1, 1, 1]]])
+    if idx < binary_op_data_shape.shape[0]:
+        l_shape = binary_op_data_shape[idx][0]
+        r_shape = binary_op_data_shape[idx][1]
+    else:
+        # Generate random data that has ndim between 1-7 and all the shape dims between 1-5
```
Comment (piiswrong) [inline tests/python/unittest/test_operator.py]: Is this enough to cover all kernels?

Comment (ap-hynninen) [inline tests/python/unittest/test_operator.py]: Yes

**Pull request merged**